versioned_hdf5/wrappers.py

```diff
@@ -595,9 +595,6 @@ def __getitem__(self, args, new_dtype=None):
         # This boilerplate code is based on h5py.Dataset.__getitem__
         args = args if isinstance(args, tuple) else (args,)
 
-        if new_dtype is None:
-            new_dtype = getattr(self._local, 'astype', None)
-
```

Comment:
Looks like there was a change in h5py which removed the same code you are removing:
https://github.com/h5py/h5py/commit/dcc694b2809c94cfa4f9718eedba5d61ab55318a
According to the comment above

> This boilerplate code is based on h5py.Dataset.__getitem__

this looks good.